File: 66-plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int i = digits.length-1;
        digits[i]++;

        while(i > 0 && digits[i] == 10 ){
                digits[i] = 0;
                digits[i-1]++;
                i--;
        }
        if(digits[0]==10){
            int result[] = new int[digits.length + 1];
            result[0] = 1;
            return result;
        }
    return digits;      
    }
}

File: 66-plus-one/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        i = len(digits) - 1
        digits[i] += 1

        while i > 0 and digits[i] == 10:
            digits[i] = 0
            digits[i - 1] = digits[i - 1] + 1
            i -= 1

        if digits[0] == 10:
            digits[0] = 0
            return [1] + digits

        return digits
